web/modules/custom/joinup_front_page/src/Plugin/Block/ExploreItemBlock.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_front_page\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides an explore item block.
+ *
+ * @Block(
+ *   id = "joinup_front_page_explore_item",
+ *   admin_label = @Translation("Explore item")
+ * )
+ */
+class ExploreItemBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Node view mode.
+   *
+   * @var string
+   */
+  protected $viewMode = 'explore_item';
+
+  /**
+   * The entity manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * The messenger.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * Constructs a new ExploreItemBlock instance.
+   *
+   * @param array $configuration
+   *   The plugin configuration, i.e. an array with configuration values keyed
+   *   by configuration option name. The special key 'context' may be used to
+   *   initialize the defined contexts by setting it to an array of context
+   *   values keyed by context names.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_manager
+   *   The entity manager.
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   The messenger.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_manager, MessengerInterface $messenger) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityManager = $entity_manager;
```

Comment:
In Drupal 8 the `EntityManager` is deprecated, and is replaced with `EntityTypeManager`. In Drupal 9 the `EntityManager` is completely removed. Since we are going to move to D9 soon it's a good idea to already adopt its practices where possible. This will minimize the number of instances where we will have to replace deprecated terminology.